attach.go

```diff
@@ -0,0 +1,155 @@
+package mail
+
+import (
+	"encoding/base64"
+	"errors"
+	"io/ioutil"
+	"mime"
+	"path/filepath"
+)
+
+// File represents the file that can be added to the email message.
+// You can add attachment from file in path, from base64 string or from []byte.
+// You can define if attachment is inline or not.
+type File struct {
+	// FilePath is the path of the file to attach.
+	FilePath string
+	// Name is the name of file in attachment. Required for Data and B64File. Optional for FilePath.
+	Name string
+	// MimeType of attachment. If empty then is obtained from Name (if not empty) or FilePath. If cannot obtained, application/octet-stream is set.
+	MimeType string
+	// B64File is the base64 string of file to attach.
+	B64File string
```

Comment:
I think this could be `B64Data` as it relates more to the `Data` field than `FilePath` field (or the former `file` argument)...

Reply:
Yes, B64Data is better. But Data refers to []byte in this case. Maybe B64String?

Reply:
Could be, but I think of it more from the perspective of the meaning of the name than the underlying data structure. Both `Data` and `B64Data` would contain the _actual_ attachment data and `FilePath` has only the name of the location of the actual data.

It is semantics and not terribly important to the actual implementation, but to my ear it sounds better to have `Data` in both field names.

Reply:
>  I think of it more from the perspective of the meaning of the name than the underlying data structure

I got it.